share/functions/pushd.fish

```diff
@@ -76,6 +76,10 @@ function pushd --description 'Push directory to stack'
     end
 
     # argv[1] is a directory
-    set -g -p dirstack $PWD
-    cd $argv[1]
+    if fish -c "cd $argv[1]" >/dev/null 2>&1
```

Comment:
That's a NAK.

We should not call fish from fish except in special circumstances.

Just check the status of `cd` before adding it to the stack

Reply:
Yeah, sorry, I realized this could be done by adding an `and` just now. Resolved.